### PR TITLE
Docs: Created a README.md for Tier 0 and minor edits

### PR DESCRIPTION
**Pull request by natalialuzuriaga**
## Problem

We should have a README.md to explain what Tier 0 is. Other tiers have this but not Tier 0 at the moment.

## Solution

- Created a README.md explaining the characteristics and required/recommended files of Tier 0.
- Renamed to Core Team, this is a recommended section so commented it out
- Renamed section to Local Development, this is a recommended section so commented it out

### CODE REVIEW

File: tier0/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+## Files for a Tier 0 Project
+
+Although these projects are private, there are specific files that are required and recommended to include in the repository as part of the CMS Open Source Program Office's repository hygiene guidelines and standards.
+
+| **File**              | **Requirement** | **Description**                                                                                             |
+|-----------------------|-----------------|-------------------------------------------------------------------------------------------------------------|
+| `README.md`           |  Mandatory     | Provides an overview of the project, including its purpose, setup instructions, or any relevant notes for the developer(s). |
+| `LICENSE`             | Mandatory        | Defines the licensing terms under which the project is distributed. |
+| `CONTRIBUTING.md`     | Recommended        | Guidelines for contributing, useful if the project is later opened to collaborators or transitioned to a public repository. |
+
```
Comment (decause-gov) [inline tier0/README.md:24]: Let's recommend SECURITY.md here as well.

File: tier0/{{cookiecutter.project_slug}}/README.md
Comment (decause-gov) [inline tier0/{{cookiecutter.project_slug}}/README.md:1]: Let's add commented out SECURITY.md policy blurb?

**Review by decause-gov — APPROVED** [review]
Comments, then LGTM +1 🚢

File: tier0/{{cookiecutter.project_slug}}/README.md
Comment (natalialuzuriaga) [reply tier0/{{cookiecutter.project_slug}}/README.md:1]: Agreed, will ship in another PR

**Pull request merged**